leapp/reporting/__init__.py

```diff
@@ -0,0 +1,193 @@
+from leapp.models import Model, fields
+from leapp.topics import Topic
+
+
+class ReportTopic(Topic):
+    name = 'report_topic'
+
+
+class ReportVersions(Model):
+    topic = ReportTopic
+
+    framework = fields.String()
+    actors = fields.String()
+
+
+class Renderers(Model):
+    topic = ReportTopic
+
+    html = fields.String()
+    plaintext = fields.String()
+
+
+class Report(Model):
+    topic = ReportTopic
+
+    """
+    Version of framework and actors as :class:`ReportVersions`
+    """
+    versions = fields.Model(ReportVersions)
+
+    """
+    Severity of the report entry
+    """
+    severity = fields.String(choices=['low', 'medium', 'high'])
+
+    """
+    Title of the report entry
+    """
+    title = fields.String()
+
+    """
+    Detail of the report entry as JSON data
+    """
+    detail = fields.String()
+
+    """
+    Named variables, represented as JSON data, that can be used during rendering
+
+    Example:
+        named = {'summary_with_exclamations': '!!!!! { summary } !!!!'}
+    """
+    named = fields.Nullable(fields.String())
+
+    """
+    :class:`Renderers` describe how to render this report entry
+    """
+    renderers = fields.Model(Renderers)
+
+    """
+    Who is the main audience of this report entry
+    """
+    audience = fields.List(fields.String(choices=['developer', 'sysadmin']))
+
+    """
+    FIXME: Do we need this here?
+    """
+    stage = fields.String()
+
+    """
+    Flags can give a special meaning to each report entry, currently supported flags:
+
+        inhibitor - This report entry will inhibit the upgrade process
+    """
+    flags = fields.List(fields.String(choices=['inhibitor']))
+
+
+"""
+Detail requirements:
+    summary : str
+
+Example:
+    {'summary': 'Some text here'}
+"""
+RenderGeneric = {
+    'html': '<h1 class="title">{ title }</h1><p class="summary">{ summary }</p>',
```

Comment:
I would use here probably h2 in all cases, because I guess that al those messages will be printed in one page that would contain `h1` header in the top of the page.

```suggestion
    'html': '<h2 class="title">{ title }</h2><p class="summary">{ summary }</p>',
```